network/vif-route-qubes

```diff
@@ -112,38 +112,58 @@ case "$command" in
         ;;
 esac
 
-domid=${vif/vif/}
-domid=${domid/.*/}
+[[ $vif =~ ^vif([0-9]+)\.([0-9]+)$ ]] || exit 1
+domid=${BASH_REMATCH[1]} sub=${BASH_REMATCH[2]}
 # metric must be possitive, but prefer later interface
 #  32752 is max XID aka domid
 metric=$(( 32752 - domid ))
 
-
+# add anti-spoofing rules before enabling the interface
 if [ "${ip}" ]; then
     # If we've been given a list of IP addresses, then add routes from dom0 to
     # the guest using those addresses.
     for addr in ${ip};
     do
-        if [[ "$addr" = *:* ]] && [[ "$ipv6_disabled" = 1 ]]; then
-            log error "Cannot set IPv6 route to ${addr}, IPv6 disabled in the kernel"
-            continue
-        fi
-        ${cmdprefix} ip route "${ipcmd}" "${addr}" dev "${vif}" metric "$metric"
         if [[ "$addr" = *:* ]]; then
             ipt=ip6tables-restore
         else
             ipt=iptables-restore
         fi
-        echo -e "*raw\\n$iptables_cmd -i ${vif} ! -s ${addr} -j DROP\\nCOMMIT" | \
```

Comment:
Hmm, I wonder how it ever worked. This filters out link-local traffic, including neighbor discovery. Looking at my firewall, Indeed this rule dropped quite a lot of packets. Perhaps it worked only thanks to periodic "neighbor advertisement" sent?

Reply:
Perhaps.  That would explain why I kept having hangs with IPv6.  Adding the permanent neighbor entries should solve the problem, however.  A follow-up PR will drop all ARP and NDP traffic in both directions, since it is an attack vector and serves no purpose.

Ideally, Xen netfront and netback interfaces would be layer 3 point-to-point links, not layer 2 Ethernet links, but changing that will be a lot more work.

Reply:
Adding permanent neighbor is fine. But dropping ARP and NDP needs some more care - we do support running other systems, including without this package installed. This applies to Windows and other systems in HVM, but also to MirageOS or other unikernels (where you don't have stubdomain to handle ARP/NDP in). Anyway, that's discussion for the other PR.